src/Raven.Studio/typescript/common/shell/accessManager.ts

```diff
@@ -1,39 +1,121 @@
 ﻿/// <reference path="../../../typings/tsd.d.ts"/>
+import activeDatabaseTracker = require("common/shell/activeDatabaseTracker");
 
 class accessManager {
 
     static default = new accessManager();
     
     static clientCertificateThumbprint = ko.observable<string>();
     
+    static databasesAccess: dictionary<Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess> = {};
+    
     securityClearance = ko.observable<Raven.Client.ServerWide.Operations.Certificates.SecurityClearance>();
 
     private allLevels = ko.pureComputed(() => true);
     
     // cluster node has the same privileges as cluster admin
     clusterAdminOrClusterNode = ko.pureComputed(() => this.securityClearance() === "ClusterAdmin" || this.securityClearance() === "ClusterNode");
         
+    isUserClearance = ko.pureComputed<boolean>(() => this.securityClearance() === "ValidUser");
+    
     operatorAndAbove = ko.pureComputed(() => { 
          const clearance = this.securityClearance();
          return clearance === "ClusterAdmin" || clearance === "ClusterNode" || clearance === "Operator";
+    });    
+    
+    getDatabaseAccessLevel(dbName: string) {
+        if (this.operatorAndAbove()) {
+            return "Admin";
+        }
+        
+        return accessManager.databasesAccess[dbName];
+    }
+
+    getDatabaseAccessLevelText(dbName: string) {
+        const accessLevel = this.getDatabaseAccessLevel(dbName);
+        if (accessLevel) {
+            switch (accessLevel) {
+                case "Admin":
+                    return "Admin access";
+                case "ReadWrite":
+                    return "Read/Write access";
+                case "Read":
+                    return "Read Only access";
+            }
+        }
+        
+        return null;
+    }
+    
+    activeDatabaseAccessLevel = ko.pureComputed<Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess>(() => {
+        const activeDatabase = activeDatabaseTracker.default.database();
+        if (activeDatabase) {
+            return this.getDatabaseAccessLevel(activeDatabase.name);
+        } 
+        
+        return null;
     });
     
+    isReadOnlyAccess = ko.pureComputed(() => this.activeDatabaseAccessLevel() === 'Read');
+    
+    canHandleOperation(requiredAccess: Raven.Client.ServerWide.Operations.Certificates.DatabaseAccess)  {
```

Comment:
will it work properly when I pass access: 'Read' ?

Reply:
fixed